frontend/src/metabase-lib/expressions/index.ts

```diff
@@ -300,19 +338,13 @@ export function isDimension(expr: unknown): boolean {
 
 export function isMetric(expr: unknown): boolean {
   return (
-    Array.isArray(expr) &&
-    expr[0] === "metric" &&
-    expr.length === 2 &&
-    typeof expr[1] === "number"
+    Array.isArray(expr) && expr[0] === "metric" && typeof expr[1] === "number"
```

Comment:
MLv2 returns 3 parameters

Reply:
Since you remove last options in `legacyFieldRef`, do we need this?

Reply:
I will revert it back, but the check looks weird to me to be honest

Reply:
https://github.com/metabase/metabase/pull/36568/commits/a5d0076d84e40bff6878ac5bc1e1c536347bf10f